packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_web.dart

```diff
@@ -2,25 +2,305 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html' as html;
+
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
+import '../../../../../primitives/auto_dispose.dart';
+import '../../../../../primitives/auto_dispose_mixin.dart';
+import '../../../../../primitives/trace_event.dart';
+import '../../../../../primitives/utils.dart';
+import '../../../../../shared/globals.dart';
+import '_perfetto_controller_web.dart';
 import 'perfetto_controller.dart';
 
-class Perfetto extends StatelessWidget {
+class Perfetto extends StatefulWidget {
   const Perfetto({
     Key? key,
     required this.perfettoController,
   }) : super(key: key);
 
   final PerfettoController perfettoController;
 
+  @override
+  State<Perfetto> createState() => _PerfettoState();
+}
+
+class _PerfettoState extends State<Perfetto> with AutoDisposeMixin {
+  late final PerfettoControllerImpl _perfettoController;
+
+  late final _PerfettoViewController _viewController;
+
+  @override
+  void initState() {
+    super.initState();
+    _perfettoController = widget.perfettoController as PerfettoControllerImpl;
+    _viewController = _PerfettoViewController(_perfettoController)..init();
+
+    _loadActiveTrace();
+    addAutoDisposeListener(
+      _perfettoController.activeTraceEvents,
+      _loadActiveTrace,
+    );
+
+    _scrollToActiveTimeRange();
+    addAutoDisposeListener(
+      _perfettoController.activeScrollToTimeRange,
+      _scrollToActiveTimeRange,
+    );
+  }
+
+  void _loadActiveTrace() {
+    unawaited(
+      _viewController._loadTrace(_perfettoController.activeTraceEvents.value),
+    );
+  }
+
+  void _scrollToActiveTimeRange() {
+    unawaited(
+      _viewController._scrollToTimeRange(
+        _perfettoController.activeScrollToTimeRange.value,
+      ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
     return Container(
-      color: Colors.white,
+      color: Theme.of(context).scaffoldBackgroundColor,
       child: HtmlElementView(
-        viewType: perfettoController.viewId,
+        viewType: _perfettoController.viewId,
       ),
     );
   }
+
+  @override
+  void dispose() {
+    _viewController.dispose();
+    super.dispose();
+  }
+}
+
+class _PerfettoViewController extends DisposableController
```

Comment:
the code in this class was all moved from PerfettoController. The one area that should be looked as new is the init method - it was partially moved from PerfettoController.